File: lib/dto/lesson.dart
class Lesson {
  final String name;
  final String description;
  final int id;
  final int totalWords;
  final int wordsLearned;

  Lesson(
      {required this.id,
      required this.name,
      required this.description,
      this.totalWords = 0,
      this.wordsLearned = 0});
}

File: app/models/Lesson.ts
import { Realm } from 'realm';

export class Lesson extends Realm.Object {
  id!: number;
  stageId!: number;
  name!: string;
  isComplete = false;

  static schema = {
    name: "Lesson",
    properties: {
      id: "int",
      stageId: "int",
      totalWords: { type: "int", default: 0 },
      totalWordsLearned: { type: "int", default: 0 },
      name: "string",
    },
  };
}
